.pre-commit-config.yaml

```diff
@@ -0,0 +1,22 @@
+repos:
+-   repo: https://gitlab.com/pycqa/flake8
+    rev: 3.8.1
+    hooks:
+    - id: flake8
+      pass_filenames: true
+      # this seems to need to be here in addition to setup.cfg
+      exclude: cibuildwheel/resources/
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v2.3.0
+    hooks:
+    - id: check-json
+    - id: check-yaml
+    - id: check-merge-conflict
+    - id: end-of-file-fixer
```

Comment:
What do these do? We're not checking them on CI either.

Reply:
yaml is checked. If there is syntax bug then CI will not start.  Other I think that are nice.

Reply:
But they aren't check on the CI. So if I don't use this pre-commit hook (or someone else), it will be triggered for you (or someone else) using the pre-commit hook?
Or is it only for new changes, and not for all files? That would be ok for me :-)

Reply:
It check only changed files. 
I do not see other yaml files than CI configuration.

Reply:
> It check only changed files.

OK, good. Then I don't really mind too much!

> I do not see other yaml files than CI configuration.

No, but there's also `check-json`. Not sure we have that much json.

Reply:
I remove json check

Reply:
No, it can stay as well, if useful to you. I'm just thinking how to keep these checks as much equal to the ones running in CI.

Reply:
In this project there is no configuration files stored in json. So their wrong formatting does not make CI fail. I copy this list from other project where this is important.